lib/ansible/modules/cloud/amazon/ssm_send_command.py

```diff
@@ -0,0 +1,216 @@
+  - Synchronous invocation may result in a function timeout, resulting in an
+    empty C(output) key.
+options:
+  name:
+    description:
+      - This should match the name of the SSM document to be invoked.
+    required: true
+    default: NONE
+  comment:
+    description:
+      - A comment about this particular invocation.
+    required: false
+    default: NONE
+  instanceIds:
+    description:
+      - This should contain an array of instance IDs for the instances you
+        wish to run this command document against.
+    required: true
+    default: NONE
+  wait:
+    description:
+      - Whether to wait for the function results or not. If I(wait) is false,
+        the task will not return any results. To wait for the command to
+        complete, set C(wait=true) and the result will be available in the
+        I(output) key.
+    required: false
+    default: true
+  parameters:
+    description:
+      - A dictionary to be provided as the parameters for the SSM document
+        you're invoking.
+    required: false
+    default: {}
+'''
+
+EXAMPLES = '''
+- ssm_send_command:
+    name: AWS-RunPowerShellScript
+    comment: "Run inventory script"
+    instanceIds:
+      - i-123987193812
+      - i-289189288278
+    parameters:
+      commands:
+        - "c:\scripts\get-inventory.ps1"
+        - "c:\scripts\cleanup.ps1"
+      workingDirectory:
+        - "c:\scripts"
+      executionTimeout:
+        - "600"
+    wait: true
+  register: response
+'''
+
+RETURN = '''
+output:
+    description: If wait=true, will return the status of sent command.
+    returned: success
+    type: dict
+    sample: "{ 'output': 'something' }"
+status:
+    description: C(Status) of AWS SSM API call
+    type: str
+    sample: Success
+'''
+
+# import base64
+# import json
+import traceback
+from time import sleep
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name                 = dict(),
+        wait                 = dict(choices=BOOLEANS, default=True, type='bool'),
+        comment              = dict(),
+        instanceIds          = dict(default=[], type='list'),
+        parameters           = dict(default={}, type='dict')
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    document_name = module.params.get('name')  # Needs to be an existing SSM document name
+    comment = module.params.get('comment')
+    await_return = module.params.get('wait')
+    instance_ids = module.params.get('instanceIds')
+    parameters = module.params.get('parameters')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    if not (document_name and instance_ids):
```

Comment:
These parameters are set to required=True however I found that if you specify the parameter in a playbook but leave it blank, it still gets past that check.  I put this in as a more friendly way of handling the exception than letting the exception generated by the the boto call to bubble up. If there is a more "Ansible-y" way of handling this, let me know. Thanks!